src/braket/circuits/instruction.py

```diff
@@ -89,6 +90,36 @@ def to_ir(self):
         """
         return self._operator.to_ir([int(qubit) for qubit in self._target])
 
+    @classmethod
+    def from_ir(cls, ir_instruction) -> Instruction:
+        """
+        Create an Instruction object from an IR instruction by calling
+        the same subclass method to implement.
+
+        Args:
+            ir_instruction: The IR instruction to create the Instruction object from
+
+        Returns:
+            Instruction: The instruction object created
+        """
+        type_name = type(ir_instruction).__name__
+        op_class = None
+        if hasattr(Gate, type_name):
+            op_class = getattr(Gate, type_name)
+        elif hasattr(Noise, type_name):  # pragma: no branch
+            op_class = getattr(Noise, type_name)
```

Comment:
Agree. Initially I had an else raising an error if `type_name` is neither a Gate or Noise. However, Coverage gave me a "miss" on the error raising line as it is never executed. Any suggestion to xfail test (with invalide `type_name`) in this case?

Reply:
I have extended `InstructionOperator = Union[Gate, Noise]` and use `InstructionOperator.__args__` to iterate through instead of hardcoding Gate and Noise. If there're no matches, it will raise a `ValueError`.

A new unit test has been created under `test_instruction_from_ir.py` to loop through all types from the schemas asserting the type names: `from braket.ir.jaqcd.program_v1 import Results, _valid_gates, _valid_noise_channels`.